ahcore/transforms/pre_transforms.py

```diff
@@ -90,23 +89,31 @@ def for_segmentation(
         if not multiclass:
             transforms.append(OneHotEncodeMask(index_map=data_description.index_map))
 
-        return cls(transforms)
+        return cls(transforms, data_description, requires_target)
 
     @classmethod
     def for_wsi_classification(
         cls, data_description: DataDescription, requires_target: bool = True
     ) -> PreTransformTaskFactory:
         transforms: list[PreTransformCallable] = []
+
+        transforms.append(SampleNFeatures(n=1000))
+
         if not requires_target:
-            return cls(transforms)
+            return cls(transforms, data_description, requires_target)
 
         index_map = data_description.index_map
         if index_map is None:
             raise ConfigurationError("`index_map` is required for classification models when the target is required.")
 
+        label_keys = data_description.label_keys
```

Comment:
Rewrite this in a more factory design, check how this is done with losses and metrics

Reply:
I've improved the function and added docstring, but I am not sure how I would improve on the factory design more.